Week1/js3-hw-w1/app.js

```diff
@@ -0,0 +1,116 @@
+'user strict';
+
+
+const url = "https://api.github.com/orgs/HackYourFuture/repos?per_page=100";
+
+function fetchJSON(url, cb) {
+
+    const request = new XMLHttpRequest();
+
+    request.open("GET", url);
+    request.responseType = "json";
+
+    request.onreadystatechange = () => {
+        if (request.readyState === 4) {
+            if (request.status < 400) {
+                cb(null, request.response);
+            } else {
+                cb(new Error(request.statusText));
+            }
+        }
+    };
+
+    request.send();
+}
+
+
+function callback(error, data) {
+    if (error !== null) {
+    } else {
```

Comment:
You didn't handle the error! If the server is unreachable, the page doesn't load at all. This isn't the best thing to do as the user would be wondering what happened. You should put text on the page saying that there was an error retrieving the data from the server.